src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -60,6 +62,67 @@ class DataSource(graphFactory: OrientGraphFactory) extends LazyLogging {
 }
 
 class RawlsTransaction(graph: OrientGraph, dataSource: DataSource) {
+
+  var readLocks:mutable.Map[Element, Int] = mutable.Map[Element, Int]().withDefaultValue(0)
```

Comment:
It sounded like you wanted a threadsafe map and triemap is pretty bad ass for those purposes